sphinx_gallery/gen_rst.py

```diff
@@ -1487,3 +1498,141 @@ def save_rst_example(
     # in case it wasn't in our pattern, only replace the file if it's
     # still stale.
     _replace_md5(write_file_new, mode="t")
+
+
+def _get_class(gallery_conf, key):
+    what = f"sphinx_gallery_conf[{repr(key)}]"
+    val = gallery_conf[key]
+    if key == "within_subsection_order":
+        alias = (  # convenience aliases
+            "ExampleTitleSortKey",
+            "FileNameSortKey",
+            "FileSizeSortKey",
+            "NumberOfCodeLinesSortKey",
+        )
+        if val in alias:
+            val = f"sphinx_gallery.sorting.{val}"
+    if isinstance(val, str):
+        if "." not in val:
+            raise ConfigError(
+                f"{what} must be a fully qualified name string, got {val}"
+            )
+        mod, attr = val.rsplit(".", 1)
+        try:
+            val = getattr(importlib.import_module(mod), attr)
+        except Exception:
+            raise ConfigError(
+                f"{what} must be a fully qualified name string, could not import "
+                f"{attr} from {mod}"
+            )
+    if not inspect.isclass(val):
+        raise ConfigError(
+            f"{what} must be 1) a fully qualified name (string) that resolves "
+            f"to a class or 2) or a class itself, got {val.__class__.__name__} "
+            f"({repr(val)})"
+        )
+    return val
+
+
+def _get_callables(gallery_conf, key):
+    # the following should be the case (internal use only)
+    singletons = ("reset_argv", "minigallery_sort_order", "subsection_order")
+    assert key in ("image_scrapers", "reset_modules", "jupyterlite") + singletons, key
+    which = gallery_conf[key]
+    if key == "jupyterlite":
+        which = [which["notebook_modification_function"]]
+    elif key in singletons:
+        which = [which]
+    if not isinstance(which, (tuple, list)):
+        which = [which]
+    which = list(which)
```

Comment:
Just a question - what does the `list(which)` do? From above, `which` would either be a tuple or a list already..?

Reply:
Makes it mutable (`tuple` is not)

Reply:
(note that at the end of this `for` we do `which[wi] = what`, which would fail for a tuple. Moreover even if we know it's a list, we don't want to modify the one in `sphinx_gallery_config` so a local `list` is better there, too!)